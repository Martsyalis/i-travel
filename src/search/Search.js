import React, { PureComponent } from 'react';
import { connect } from 'react-redux';


class Search extends PureComponent {


  render() {
    const searchedExp = this.props.exp.find(exp => exp._id === this.props.id);
    if (!searchedExp) return <div>no such experience has been posted yet</div>;

    return (
      <div>
        <h1>Hey {this.props.user.name} Welcome to Experience page</h1>
        <h3>title is: {searchedExp.title}</h3>
        <h5>Location:  {searchedExp.location} </h5>
        {(searchedExp.images)
          ? (<ul>
            {searchedExp.images.map(image => (
              <li key={image._id}>
                <img src={image.imageURI} alt={image.caption} />
                <h5> {image.caption} </h5>
              </li>
            ))}
          </ul>)
          : <div> No images uploaded yet </div>
        }
      </div>
    );
  }
}

export default connect(
  state => ({ user: state.auth.user, exp: state.experiences }),
  {  }
)(Search); 
