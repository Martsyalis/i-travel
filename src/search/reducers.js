export const SEARCH_LOAD = 'SEARCH_LOAD';

export function search(state = [], { type, payload }) {
  switch (type) {
    case SEARCH_LOAD:
      return payload;
    default:
      return state;
  }
}
