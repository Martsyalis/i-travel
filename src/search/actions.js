import experiencesApi from '../services/experiences-Api';
import { SEARCH_LOAD } from './reducers';

export function loadSearch(query) {
  console.log('we are quering with:',query);
  return (dispatch, getState) => {
    dispatch({
      type: SEARCH_LOAD,
      payload: experiencesApi.search(query)
    });
  };
}

