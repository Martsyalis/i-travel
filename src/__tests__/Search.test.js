import React from 'react';

import { shallow } from 'enzyme';

import { Search } from '../search/Search';

describe('Search', () => {
  it('shows the Search component', () => {
    const wrapper = shallow(<Search user={{}} search={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
