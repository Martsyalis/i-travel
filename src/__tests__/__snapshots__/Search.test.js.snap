// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search shows the Search component 1`] = `
<div>
  <section
    className="hero is-dark"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Hey 
        </h1>
        <h2
          className="subtitle"
        >
          Please enter your Search
        </h2>
      </div>
    </div>
  </section>
  <div
    className="tile"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          className="label"
        >
          Location
        </label>
        <div
          className="control"
        >
          <input
            name="location"
            placeholder="location"
          />
        </div>
      </div>
      <div
        className="field"
      >
        <label
          className="label"
        >
          Tag
        </label>
        <div
          className="control"
        >
          <input
            name="tag"
            placeholder="tag"
          />
        </div>
      </div>
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div>
     No results 
  </div>
</div>
`;
